### PR TITLE
Add default AppBar for use across pages

### LINKED ISSUE #48: Factor out the default AppBar so that it can be reused across pages.

### PR DESCRIPTION
**Pull request by KoopaKing**
Fixes #48

### CODE REVIEW

File: frontend/ui/test/widgets/default_app_bar_test.dart [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+void main() {
+  testWidgets('displays expected title', (WidgetTester tester) async {
+    await tester.pumpWidget(_InTestableContext(defaultAppBar));
+
+    expect(find.text('TripMeOut'), findsOneWidget);
+  });
+}
+
+typedef _AppBarProvider = Widget Function(BuildContext);
```
Comment (bsidhom) [inline frontend/ui/test/widgets/default_app_bar_test.dart:13]: Is this typedef for readability or does dart not have a way to declare an in-line function type?

**Review by bsidhom — APPROVED** [review]

File: frontend/ui/test/widgets/default_app_bar_test.dart (hunk 1 above)
Comment (KoopaKing) [reply frontend/ui/test/widgets/default_app_bar_test.dart:13]: for readability.

Comment (KoopaKing) [reply frontend/ui/test/widgets/default_app_bar_test.dart:13]: as far as I know it's like go's typedef

**Review by anfee1 — APPROVED** [review]

File: frontend/ui/test/widgets/default_app_bar_test.dart (hunk 1 above)
Comment (bsidhom) [reply frontend/ui/test/widgets/default_app_bar_test.dart:13]: +1 for readability. Sadly, looks like this is actually [more like](https://dart.dev/guides/language/language-tour#typedefs) Go type aliases. Go typedefs actually mint new mutually incompatible types (modulo explicit casts).

**Pull request merged**